lib/likes.ex

```diff
@@ -195,18 +192,41 @@ defmodule Bonfire.Social.Likes do
   end
 
   def do_like(%{} = liker, %{} = liked, opts \\ []) do
+    debug("do_like: liker=#{id(liker)}, liked=#{inspect(liked.__struct__)}, id=#{id(liked)}")
+
     liked = Objects.preload_creator(liked)
     liked_object_creator = Objects.object_creator(liked)
 
-    opts =
+    # Fallback: if no creator found, try to get it from activity or other associations
+    liked_object_creator =
+      if is_nil(liked_object_creator) do
+        # Check if we're dealing with an Activity and need to get its object's creator
+        case liked do
+          %Activity{} = activity ->
+            activity = repo().maybe_preload(activity, object: [created: [creator: :character]])
+            e(activity, :object, :created, :creator, nil) || e(activity, :subject, nil)
+
+          _ ->
+            # Try different paths to find the creator for non-Activity objects
+            e(liked, :activity, :subject, nil) ||
+              e(liked, :created, :creator, nil) ||
+              e(liked, :post_content, :created, :creator, nil)
+        end
+      else
+        liked_object_creator
+      end
+
+    debug("do_like: found creator #{inspect(id(liked_object_creator))}")
+
+    new_opts =
```

Comment:
no need to call it something else, put actually would be cleaner to have opts = opts |> Keyword.merge(...